clang/test/CIR/CodeGen/array.cpp

```diff
@@ -28,14 +28,99 @@ int f[5] = {1, 2};
 // CHECK: cir.global external @f = #cir.const_array<[#cir.int<1> : !s32i, #cir.int<2> : !s32i, #cir.int<0> : !s32i, #cir.int<0> : !s32i, #cir.int<0> : !s32i]> : !cir.array<!s32i x 5>
 
 void func() {
-  int l[10];
-  // CHECK: %[[ARR:.*]] = cir.alloca !cir.array<!s32i x 10>, !cir.ptr<!cir.array<!s32i x 10>>, ["l"]
+  int arr[10];
+
+  // CHECK: %[[ARR:.*]] = cir.alloca !cir.array<!s32i x 10>, !cir.ptr<!cir.array<!s32i x 10>>, ["arr"]
+}
+
+void func2() {
+  int arr[2] = {5};
```

Comment:
Can you add tests with non-int types? I'm particularly interested in how an array of pointers would be zero-initialized.